redisvl/index.py

```diff
@@ -371,61 +486,94 @@ def delete(self, drop: bool = True):
         self._redis_conn.ft(self._name).dropindex(delete_documents=drop)  # type: ignore
 
     @check_connected("_redis_conn")
+    @check_modules_present("_redis_conn")
```

Comment:
Can we use ``lru_cache`` to cache the calls to this within an invocated client?

Reply:
hmm i've not done that. are you saying cache the module list client side? I was going to recommend some kind of private attribute on the class like `_modules` that gets populated first time you run it. Then next time the check is called, it looks at this attribute first, and then executes the redis command. Something like that. Lmk what you're thinking? Feels like a later optimization to me